raiden/transfer/utils.py

```diff
@@ -77,13 +83,21 @@ def encrypt_secret(
     public_key = get_public_key(message, signature)
     encrypted_secret = None
     if public_key:
-        encrypted_secret = EncryptedSecret(encrypt(public_key.to_hex(), secret))
+        to_encrypt = {
+            "secret": encode_hex(secret),
+            "amount": amount,
+            "payment_identifier": payment_identifier,
+        }
+        encrypted_secret = EncryptedSecret(
+            encrypt(public_key.to_hex(), json.dumps(to_encrypt).encode())
+        )
     return encrypted_secret
 
 
 def decrypt_secret(encrypted_secret: EncryptedSecret, private_key: PrivateKey) -> Secret:
     try:
-        secret = Secret(decrypt(private_key, encrypted_secret))
-    except ValueError:
+        secret_dict = json.loads(decrypt(private_key, encrypted_secret).decode())
+        secret = Secret(decode_hex(secret_dict["secret"]))
```

Comment:
Are you validating `amount` and `payment_identifier` too? At least the first is required, since target knowing the secret means it won't ask initiator for it, target is the part responsible for ensuring mediators didn't take more than expected. If you just use the secret, mediators could rob target and no one would notice, so this is a security flaw.
clients should assume that, if target did unlock the transfer, they've agreed they've received at least the amount intended (`secret.amount`), but then this must be validated strictly.
`payment_identifier` was also previously validated by initiators, and although it's just a metadata, it _may_ contain sensible information which we should ensure this is correct.

Reply:
Notice also that the amount and payment_identifer, as per the rules of Raiden, are serialized as strings.

Reply:
Yes, this is being validated when we process the transaction. We already have that info in the current state of the node.